frontend/src/Pages/FacilityManagement.tsx

```diff
@@ -40,73 +46,104 @@ export default function FacilityManagement() {
     } = useSWR<ServerResponseMany<Facility>, AxiosError>(
         `/api/facilities?page=${pageQuery}&per_page=${perPage}`
     );
-    const { revalidate } = useRevalidator();
 
     const facilityData = facility?.data ?? [];
 
-    function resetModal() {
-        setTimeout(() => {
-            setEditFacility(undefined);
-        }, 200);
-    }
+    useEffect(() => {
+        if (targetFacility?.action == CRUDActions.Edit) {
+            editFacilityModal.current?.showModal();
+        }
+        if (targetFacility?.action == CRUDActions.Add) {
+            addFacilityModal.current?.showModal();
+        }
+        if (targetFacility?.action == CRUDActions.Delete) {
+            deleteFacilityModal.current?.showModal();
+        }
+    }, [targetFacility]);
+
+    const openDeleteFacility = (facility: Facility) => {
+        setTargetFacility({
+            action: CRUDActions.Delete,
+            facility: facility
+        });
+    };
```

Comment:
It feels like these should have the word handle in the name. My initial reaction was that it's doing the same as what is in the `useEffect`